File: 백준/Gold/11404. 플로이드/플로이드.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.StringTokenizer;

public class Main {

    static int n, m;
    static long[][] graph;
    static final int INF = Integer.MAX_VALUE;

    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

        n = Integer.parseInt(br.readLine()); // 도시의 개수
        m = Integer.parseInt(br.readLine()); // 버스의 개수

        graph = new long[n + 1][n + 1];

        for (int i = 1; i <= n; i++) {
            for (int j = 1; j <= n; j++) {
                if (i == j) graph[i][i] = 0;
                else graph[i][j] = INF;
            }
        }

        while (m-- > 0) {
            StringTokenizer st = new StringTokenizer(br.readLine());
            int start = Integer.parseInt(st.nextToken());
            int end = Integer.parseInt(st.nextToken());
            int weight = Integer.parseInt(st.nextToken());

            if (graph[start][end] > weight) {
                graph[start][end] = weight;
            }
        }

        for (int k = 1; k <= n; k++) {
            for (int i = 1; i <= n; i++) {
                for (int j = 1; j <= n; j++) {
                    graph[i][j] = Math.min(graph[i][j], graph[i][k] + graph[k][j]);

                }
            }
        }


        for (int i = 1; i <= n; i++) {
            for (int j = 1; j <= n; j++) {
                if (graph[i][j] == INF) {
                    System.out.print(0 + " ");
                } else {
                    System.out.print(graph[i][j] + " ");
                }
            }
            System.out.println();
        }

    }

}

File: 백준/Gold/11404. 플로이드/플로이드.py
import sys
input = sys.stdin.readline

n = int(input())  # 도시의 개수
m = int(input())  # 노선의 개수

inf = sys.maxsize

distance = [[inf for _ in range(n + 1)] for _ in range(n + 1)]
for i in range(1, n + 1):
    distance[i][i] = 0


for _ in range(m):
    start, end, weight = map(int, input().split())
    if distance[start][end] > weight:
        distance[start][end] = weight

for k in range(1, n + 1):
    for i in range(1, n + 1):
        for j in range(1, n + 1):
            distance[i][j] = min(distance[i][j], distance[i][k] + distance[k][j])

for i in range(1, n + 1):
    for j in range(1, n + 1):
        if distance[i][j] == inf:
            print(0, end=" ")
        else:
            print(distance[i][j], end=" ")
    print()





